### DISCUSSION TITLE
Can I call async functions in user defined functions

### DISCUSSION THREAD

**Discussion by strazto** [agershun/alasql#1690] [Q&A] [answered]
According to the wiki: https://github.com/AlaSQL/alasql/wiki/User-Defined-Functions , 

> You can use alasql inside alasql functions, like below:
> ```js
> alasql.fn.myfilter = function(phase) {
> 	return alasql('SELECT VALUE COUNT(*) FROM ? WHERE Phase = ?',[data,phase]) == 2;
> };
> 
> var res = alasql('SELECT * FROM ? WHERE myfilter(Phase)',[data]);
> ```

Given that the result of `alasql()` can be a promise, this would imply that you can execute promises in user defined functions. Is that right, or if not, is there a way to make it so>

**Comment by mathiasrw** [comment, marked as answer]
The user defined function is called as a sync function, so you will have to deal with the isolating the async part within the function. 

Give me an example of what you hope to be able to do.
